.github/workflows/code.yml

```diff
@@ -64,18 +69,61 @@ jobs:
           name: ${{ matrix.python }}/${{ matrix.os }}
           files: cov.xml
 
-  container:
+      - name: Upload lockfiles
+        uses: actions/upload-artifact@v3
+        with:
+          name: lockfiles
+          path: lockfiles
+
+  dist:
     if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name != github.repository
+    runs-on: "ubuntu-latest"
+
+    steps:
+      - name: Checkout Source
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Build Sdist and wheel
+        run: |
+          export SOURCE_DATE_EPOCH=$(git log -1 --pretty=%ct) && \
+          pipx run build
+
+      - name: Test module --version works using the installed wheel
+        run: |
+          touch requirements.txt
```

Comment:
Should we use the script here too?

Reply:
We could use the script here instead of generating requirements.txt in the container.

Now that I'm installing from the WHL here (previously was using sdist) I think it makes sense to move the requirements.txt generation here. This solves the DRY issue and means we always use this script for generating the lock files.

Reply:
UPDATE: this does not work because installing the WHL into the system python means that we see all sorts of pre-installed packages that GHA uses in the requirements.txt file we generate.

I'm switching back to making requirements.txt in the container. use an explicit sed there which comments out the self reference and use a simple deletion sed in the the script (because it is used with dev dependencies and will always have multiple lines - delete will never create an empty asset)